be2-scala/src/main/scala/ch/epfl/pop/model/objects/Base64Data.scala

```diff
@@ -15,6 +15,12 @@ case class Base64Data(data: String) {
   def equals(that: Base64Data): Boolean = data == that.data
 
   override def toString: String = data.toString
+
+  try {
+    DECODER.decode(data);
+  } catch {
+    case _: IllegalArgumentException => throw new IllegalArgumentException(s"String $data is not Base64")
```

Comment:
I don't see a test that checks this case, it might be worthwhile to have one.

Reply:
Sure I'll add to one of  my pending PRs